src/Juvix/Bohm/Type.hs

```diff
@@ -23,9 +24,30 @@ data Bohm
   -- Not valid syntax but for read back of a graph
   | Erase
   -- Not valid syntax but for read back of a graph
-  | Curried (Int → Int) Bohm
-  | CurriedB (Int → Bool) Bohm
+  | Curried3 (Primitive → Primitive → Primitive → Maybe Primitive) Bohm Bohm Bohm
```

Comment:
Why `Maybe`? Should be a compiler bug if the types mismatch, right?

Reply:
it would be a compiler bug yes, but how I implement the functions currently is pattern matches on primitive to apply itself, it could throw an error on a mismatch however.